support/azureutil/azureutil.go

```diff
@@ -0,0 +1,37 @@
+package azureutil
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+)
+
+// GetSubnetNameFromSubnetID extracts the subnet name from a subnet ID
+// Example subnet ID: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>/subnets/<subnetName>
+func GetSubnetNameFromSubnetID(subnetID string) (string, error) {
+	subnet, err := arm.ParseResourceID(subnetID)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse subnet ID %q: %v", subnetID, err)
+	}
+
+	if subnet.Name == "" {
+		return "", fmt.Errorf("failed to parse subnet name from %q", subnetID)
+	}
+
+	return subnet.Name, nil
+}
+
+// GetNetworkSecurityGroupNameFromNetworkSecurityGroupID extracts the network security group (nsg) name from a nsg ID
+// Example nsg ID: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<nsgName>
+func GetNetworkSecurityGroupNameFromNetworkSecurityGroupID(nsgID string) (string, error) {
```

Comment:
This function differs very little from GetSubnetNameFromSubnetID, could we use one function for both subnet and network security groups?

Reply:
I prefer to have separate functions for different things rather than trying to squeeze them all into the same function.

Reply:
I think it makes the code easier to read and understand what it is doing.